java/templates/s2i/run

```diff
@@ -13,6 +13,45 @@ JAVA_OPTIONS="${JAVA_OPTIONS:+${JAVA_OPTIONS} }$(/opt/jolokia/jolokia-opts)"
 JVM_ARGS="${JVM_ARGS:+${JVM_ARGS} }${JAVA_OPTIONS}"
 export JAVA_OPTIONS JVM_ARGS
 
+# support JBoss Tools OpenShift debug environment variables
+export JAVA_DEBUG="${JAVA_DEBUG:-${DEBUG}}"
+if [ "x${JAVA_DEBUG}" != x ]; then
+  export JAVA_DEBUG_PORT="${JAVA_DEBUG_PORT:-${DEBUG_PORT}}"
+fi
+# SpringBoot also uses DEBUG variable, so shade it with SPRINGBOOT_DEBUG variable
+if [ "x${SPRINGBOOT_DEBUG}" != x ]; then
+	export DEBUG="${SPRINGBOOT_DEBUG}"
+else
+	unset DEBUG
+fi
+
+# check for SpringBoot exploded archive
+if [ -f "${DEPLOYMENTS_DIR}/META-INF/MANIFEST.MF" ]; then
+	echo "SpringBoot exploded archive detected..."
+	if [ "x${JAVA_DEBUG}" != x ]; then
+		# does archive container spring devtools in lib
+		devtools_present=`ls "${DEPLOYMENTS_DIR}"/BOOT-INF/lib/spring-boot-devtools-*.jar 2>/dev/null | wc -l | tr -d '[[:space:]]'`
+		if [ ${devtools_present} = 1 ]; then
+			echo "SpringBoot developer tools library detected..."
+			# setup classpath and start class as main class
+			if [ "x${JAVA_CLASSPATH}" = x ]; then
+				JAVA_CLASSPATH="${DEPLOYMENTS_DIR}/BOOT-INF/lib/*"
+			else
+				JAVA_CLASSPATH="${DEPLOYMENTS_DIR}/BOOT-INF/lib/*:${JAVA_CLASSPATH}"
+			fi
+			export JAVA_CLASSPATH="${DEPLOYMENTS_DIR}/BOOT-INF/classes/:${JAVA_CLASSPATH}"
+			# setup start class as main class
+			main_class=`sed -n 's/Start-Class: \(.*\)$/\1/p' "${DEPLOYMENTS_DIR}/META-INF/MANIFEST.MF" | tr -d '[[:space:]]'`
+		fi
+	fi
+	# default to SpringBoot's Main-Class
+	if [ "x${main_class}" = x ]; then
```

Comment:
If JAVA_DEBUG is not set or nor devtools present, then there is no classpath adoption for `BOOT-INF/lib/*` path. Is this correct ?